### DISCUSSION TITLE
Ability to display gpx/kml for pre-planned walks.

### DISCUSSION THREAD

**Discussion by InsertUser** [streetcomplete/StreetComplete#3898] [Ideas]
I sometimes find it useful to have a route pre-planned to reduce the number of dead ends I stumble into, for this I use a route service that automatically generated a route of a certain length and allows me to load that as a gpx of kml into one of my offline maps apps to follow onscreen. 

It would be really handy if I could load this as a line directly in StreetComplete to minimise the amount of app switching and "wrong turns". For my usage I wouldn't really need more that one loaded at a time as they tend to be something I'd use once and discard. 

In a perfect world it would also allow me to pre-cache quests and map tiles along the route as for JOSM's download along feature, but that's probably taking things several steps too far. 

Just wondered how many others would find this useful really. Am I the only one who does this?

**Comment by riQQ** [comment]
See also #1757

**Reply by InsertUser** [reply]
I think for me there could be less interface than mentioned there. StreetComplete would just need to be one of the options when trying to open a file and have a "clear external track" button for when you're finished. Maybe a popup to confirm that you really want to clear it?

That way it would be less noticeable for those it might confuse, but would still appear when you need it.

**Comment by mnalis** [comment]
Yes, I often (always?) run into this. 

My workaround is basically always running OsmAnd alongside for navigation and track recording (which also comes helpful later when I try resolve notes I left in SC when mapping). OsmAnd has audio navigation too so that reduces amount of switching I need to do (although I still switch, for example to verify route in bigger picture or if I didn't listen carefully, and to map/verify missing POIs I find important - mostly cycling-related ones). 

Nice shortcut in android is double-clicking on android rectangle navigation icon, which acts like single-press `alt-tab` on desktop - always switching between last two apps, so that reduces the pain too. 

I'd love  _"cache map & quests along this gpx track"_ part of the proposal though (like Vespucci [implemented recently](https://github.com/MarcusWolschon/osmeditor4android/issues/1217)), as that would allow me to disable mobile data completely while mapping, thus substantially increasing battery runtime.

**Reply by InsertUser** [reply]
I use OsmAnd for notes and tracks but it's rendering slow for me so I'm using Organic Maps for the track display. I haven't been using headphones the last few trips so I've left the voice prompts off.

My latest Android update swapped the old buttons for a bar I can swipe along to get the previous app, but the switching still breaks up the flow of things.

**Reply by mnalis** [reply]
@InsertUser just to note for interesting parties that [SCEE](https://wiki.openstreetmap.org/wiki/SCEE) _"expert edition"_ fork of StreetComplete nowadays does allow for showing the `.gpx` track, and even downloading alongside it, although downloading part needs some improvements currently (see https://github.com/Helium314/SCEE/issues/471), and there is renewed discussion and links to some more code in https://github.com/streetcomplete/StreetComplete/discussions/5345 discussion too.

**Comment by drazde** [comment]
I also think it could be useful. Usually at the end of a mountain bike excursion I check if the surfaces are all classified, otherwise I map them.
If I could upload the tracks it would be easier
